docs/SimpleFormIterator.md

```diff
@@ -150,18 +150,17 @@ const PostEdit = () => (
 **Note**: `<SimpleFormIterator>` only accepts `Input` components as children. If you want to use some `Fields` instead, you have to use a `<FormDataConsumer>` to get the correct source, as follows:
 
 ```jsx
-import { ArrayInput, SimpleFormIterator, DateInput, TextInput, FormDataConsumer } from 'react-admin';
+import { ArrayInput, SimpleFormIterator, DateInput, TextInput, FormDataConsumer, Labeled } from 'react-admin';
 
 <ArrayInput source="backlinks">
     <SimpleFormIterator disableRemove >
         <DateInput source="date" />
         <FormDataConsumer>
-            {({ getSource, scopedFormData }) => {
+            {({ getSource }) => {
                 return (
-                    <TextField
-                        source={getSource('url')}
-                        record={scopedFormData}
-                    />
+                    <Labeled label="Url">
+                        <TextField source={getSource('url')} />
```

Comment:
This is wrong, this will show the data from the record, not the form. 

If it doesn't work with scopedFormData, it may be because the usewatch() in FormDataConsumer doesn't return data before changes. This is what we must fix, not the docs.

Reply:
Might be linked to https://github.com/marmelab/react-admin/pull/8340